modnet/models.py

```diff
@@ -231,6 +232,12 @@ def fit(
                 )
                 loss = "categorical_crossentropy"
             else:
+                # This is a compatibility thing: we have some pre-featurized data that has differing
+                # column names that we would like to use. Most of this is single-target, so using just the
+                # one column name will always work. Otherwise this will raise a missing column error unless
+                # columns are renamed.
+                if len(self.targets_flatten) == 1:
+                    targ = list(training_data.df_targets.columns)[0]
```

Comment:
could also rename the precomputed columns or target_names

Reply:
maybe move this condition at the very beginning, and set targets_flatten to the columns name, will fix the next comment